src/main/java/org/wso2/carbon/securevault/hashicorp/repository/HashiCorpSecretRepository.java

```diff
@@ -0,0 +1,182 @@
+    private String token;
+    private String address;
+    private String namespace;
+    private String enginePath;
+    private int engineVersion;
+
+    public HashiCorpSecretRepository(IdentityKeyStoreWrapper identityKeyStoreWrapper,
+                                     TrustKeyStoreWrapper trustKeyStoreWrapper) {
+
+        this.identityKeyStoreWrapper = identityKeyStoreWrapper;
+        this.trustKeyStoreWrapper = trustKeyStoreWrapper;
+    }
+
+    /**
+     * Initializes the repository based on provided properties.
+     *
+     * @param properties Configuration properties
+     * @param id         Identifier to identify properties related to the corresponding repository
+     */
+    @Override
+    public void init(Properties properties, String id) {
+
+        LOG.info("Initializing HashiCorp Secure Vault");
+
+        // Load Configurations
+        HashiCorpVaultConfigLoader hashiCorpVaultConfigLoader = HashiCorpVaultConfigLoader.getInstance();
+        try {
+            address = hashiCorpVaultConfigLoader.getProperty(ADDRESS_PARAMETER);
+            namespace = hashiCorpVaultConfigLoader.getProperty(NAMESPACE_PARAMETER);
+            enginePath = hashiCorpVaultConfigLoader.getProperty(ENGINE_PATH_PARAMETER);
+
+            String version = hashiCorpVaultConfigLoader.getProperty(ENGINE_VERSION_PARAMETER);
+            engineVersion = version != null ? Integer.parseInt(version) : DEFAULT_ENGINE_VERSION;
+        } catch (HashiCorpVaultException e) {
+            LOG.error(e.getMessage(), e);
+        }
+
+        // Get the vault token
+        token = System.getenv(TOKEN_PARAMETER);
+        if (token == null || token.isEmpty()) {
+            LOG.warn("VAULT_TOKEN environment variable is not set");
+        }
+
+        if (engineVersion != 2) {
+            LOG.error("Supported engine version: 2");
+        }
+    }
+
+    /**
+     * Get Secret from the Secret Repository
+     *
+     * @param alias Alias name for look up a secret
+     * @return Secret if there is any, otherwise, alias itself
+     * @see SecretRepository
+     */
+    @Override
+    public String getSecret(String alias) {
+
+        if (StringUtils.isEmpty(alias)) {
+            return alias;
+        }
+
+        StringBuilder sb = new StringBuilder()
+                .append(enginePath)
+                .append(SLASH)
+                .append(alias);
+
+        String secret = null;
+        try {
+
+            final VaultConfig config = new VaultConfig()
+                    .address(address)
+                    .token(token)
+                    .engineVersion(engineVersion)
+                    .build();
+
+            Vault vault = new Vault(config);
+            Logical logical = vault.logical();
+            if (namespace != null) {
+                logical = logical.withNameSpace(namespace);
+            }
+            secret = logical.read(sb.toString()).getData()
+                    .get(VALUE_PARAMETER);
+
+            if (secret == null) {
```

Comment:
Use StringUtils.isEmpty for the null check and check this throughout the PR

Reply:
Resolved in #6